library/src/scripts/forms/select/SelectBox.tsx

```diff
@@ -79,36 +81,70 @@ export default class SelectBox extends React.Component<ISelfLabelledProps | IExt
             selectedIndex: index,
             selectedItem,
         });
+        this.props.handleOnChange(selectedItem.name);
     };
 
+    private renderChild = (child, selected, classes) => {
```

Comment:
This seems like something that should just be its own component. While we have done this type of thing in the past it's a bit of an anti-pattern.